compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -621,7 +621,7 @@ private void createFunctionMap(BLangFunction funcNode, SymbolEnv funcEnv) {
         if (funcNode.mapSymbolUpdated) {
             return;
         }
-        BLangRecordLiteral emptyRecord = ASTBuilderUtil.createEmptyRecordLiteral(funcNode.pos, symTable.mapType);
+        BLangRecordLiteral emptyRecord = ASTBuilderUtil.createEmptyRecordLiteral(funcNode.pos, symTable.mapAllType);
```

Comment:
@chiranSachintha, don't we have function parameter tests for this? Let's create an issue for this and add some tests.